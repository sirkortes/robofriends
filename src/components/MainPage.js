// import React, { useState, useEffect } from "react";
import React from 'react'
import CardList from '../components/CardList'
import SearchBox from '../components/SearchBox'
import Scroll from '../components/Scroll'
import ErrorBoundry from '../components/ErrorBoundry'
import './MainPage.css'

function MainPage({ isPending, error, robots, searchField }) {
  const filterRobots = () =>
    robots.filter((robot) =>
      robot.name.toLowerCase().includes(searchField.toLowerCase())
    )

  return isPending ? (
    <h1 className='tc f2'>Loading...</h1>
  ) : !!error ? (
    <div className='tc'>
      <h1 className='tc f2'>Error</h1>
      <p>{error}</p>
    </div>
  ) : (
    <div className='tc'>
      <h1 className='f2'>RoboFriends</h1>
      <SearchBox />
      <Scroll>
        <ErrorBoundry>
          <CardList robots={filterRobots()} />
        </ErrorBoundry>
      </Scroll>
    </div>
  )
}

export default MainPage
