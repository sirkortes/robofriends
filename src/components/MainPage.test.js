import { shallow } from 'enzyme'
import React from 'react'
import MainPage from './MainPage'

let wrapper
beforeEach(() => {
  const mockProps = {
    error: '',
    robots: [],
    searchField: '',
    isPending: false
  }
  wrapper = shallow(<MainPage {...mockProps} />)
})

describe('Component: MainPage', () => {
  it('renders MainPage without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  })

  // it('filters robots correctly', () => {
  // const wrapdebug = wrapper.debug()
  // const wrapperfind = wrapper.find('CardList')
  // console.log('wrapdebug is [%s]: ', typeof wrapdebug, wrapdebug)
  // console.log('wrapperfind is [%s]: ', typeof wrapperfind, wrapperfind)
  // expect(wrapperfind.filterRobots([])).toEqual([])
  //
  // starting react 16+ shallow wrapper instance is null for stateless components
  // so we cant do above kind of test
  // })
})
