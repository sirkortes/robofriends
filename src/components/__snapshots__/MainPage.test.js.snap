// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MainPage renders MainPage without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    error=""
    isPending={false}
    robots={Array []}
    searchField=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="f2"
>
          RoboFriends
</h1>,
        <SearchBox />,
        <Scroll>
          <ErrorBoundry>
                    <CardList
                              robots={Array []}
                    />
          </ErrorBoundry>
</Scroll>,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "RoboFriends",
          "className": "f2",
        },
        "ref": null,
        "rendered": "RoboFriends",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ErrorBoundry>
            <CardList
                        robots={Array []}
            />
</ErrorBoundry>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <CardList
              robots={Array []}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "robots": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="f2"
>
            RoboFriends
</h1>,
          <SearchBox />,
          <Scroll>
            <ErrorBoundry>
                        <CardList
                                    robots={Array []}
                        />
            </ErrorBoundry>
</Scroll>,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "RoboFriends",
            "className": "f2",
          },
          "ref": null,
          "rendered": "RoboFriends",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ErrorBoundry>
              <CardList
                            robots={Array []}
              />
</ErrorBoundry>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <CardList
                robots={Array []}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "robots": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSeventeenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "snapshotSerializers": Array [
      "enzyme-to-json/serializer",
    ],
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
